### PR TITLE
Board specific flash sizes for RP2040

### PR DESCRIPTION
**Pull request by tannewt**
Stop-gap solution for #4041. Comment is there to provide info
needed in the future. (We currently run the generic "safe" settings.)

### CODE REVIEW

**Review by ladyada — APPROVED** [review]

File: ports/raspberrypi/boards/adafruit_feather_rp2040/mpconfigboard.h [hunk 1]
```diff
@@ -12,3 +12,6 @@
 
 // #define DEFAULT_UART_BUS_RX (&pin_PA11)
 // #define DEFAULT_UART_BUS_TX (&pin_PA10)
+
+// Flash chip is GD25Q32 connected over QSPI
+#define TOTAL_FLASH_SIZE 4 * 1024 * 1024
```
Comment (dhalbert) [inline ports/raspberrypi/boards/adafruit_feather_rp2040/mpconfigboard.h:17]: These should be parenthesized.

File: ports/raspberrypi/boards/qtpy_rp2040/mpconfigboard.h [hunk 2]
```diff
@@ -12,3 +12,6 @@
 
 // #define DEFAULT_UART_BUS_RX (&pin_PA11)
 // #define DEFAULT_UART_BUS_TX (&pin_PA10)
+
+// Flash chip is GD25Q32 connected over QSPI
+#define TOTAL_FLASH_SIZE 4 * 1024 * 1024
```
Comment (dhalbert) [inline ports/raspberrypi/boards/qtpy_rp2040/mpconfigboard.h:17]: parens

File: ports/raspberrypi/boards/raspberry_pi_pico/mpconfigboard.h [hunk 3]
```diff
@@ -13,3 +13,6 @@
 
 // #define DEFAULT_UART_BUS_RX (&pin_PA11)
 // #define DEFAULT_UART_BUS_TX (&pin_PA10)
+
+// Flash chip is W25Q16JVUXIQ connected over QSPI
+#define TOTAL_FLASH_SIZE 2 * 1024 * 1024
```
Comment (dhalbert) [inline ports/raspberrypi/boards/raspberry_pi_pico/mpconfigboard.h:18]: parens

File: ports/raspberrypi/supervisor/internal_flash.c [hunk 4]
```diff
@@ -46,9 +46,6 @@
 
 #define RESERVED_FLASH 1 * 1024 * 1024
```
Comment (dhalbert) [inline ports/raspberrypi/supervisor/internal_flash.c:47]: parens

**Review by dhalbert — CHANGES_REQUESTED** [review]

**Comment by tannewt** [comment]
Parens added. Thanks!

**Review by dhalbert — APPROVED** [review]
Thanks for accommodating my pedantry.

**Pull request merged**